### PR TITLE
Update section about TLS

### PR DESCRIPTION
**Pull request by kwart**
This PR contains commits changing `security.adoc`.
- clean/trail line endings;
- update TLS related sections in the documentation;

### CODE REVIEW

File: src/docs/asciidoc/security.adoc [hunk 1]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
 * `trustStorePassword`: Password to unlock the truststore file.
 * `trustManagerAlgorithm`: Name of the algorithm based on which the trust managers are provided.
-* `trustStoreType`: The type of the truststore. Its default value is `JKS`. 
+* `trustStoreType`: The type of the truststore. Its default value is `JKS`.
+* `mutualAuthentication`: Mutual authentication configuration. It's empty by default which means the client side of connection is not authenticated. Available values are:
+** `REQUIRED` - server forces usage of a trusted client certificate
+** `OPTIONAL` - server asks for a client certificate, but it doesn't require it
++
+Please see the <<mutual-authentication, Mutual Authentication section>>.
+* `ciphersuites`: Comma-separated list of cipher suite names allowed to be used. Its default value are all supported suites in your Java runtime.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: Maybe including this property (with an example value) in the above <ssl> declarative configuration is a good idea?

File: src/docs/asciidoc/security.adoc [hunk 2]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+        <property name="protocol">TLSv1.2</property>
+      </properties>
+    </ssl>
+  </network>
+  ...
+</hazelcast-client>
 ```
 
+The same `BasicSSLContextFactory` properties used for members are available on clients.
+Clients don't need to set `mutualAuthentication` property as it's property configuring server side of TLS connections.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: .....property as it's used in configuring the server side of TLS connections.

File: src/docs/asciidoc/security.adoc [hunk 3]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+
 ==== Mutual Authentication
 
-As explained above, Hazelcast members have keyStore used to identify themselves (to other members) and Hazelcast clients have trustStore used to define which members they can trust. 
+TLS connections have two sides - one which opens the connection (i.e. TLS client)
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: TLC connections have two sides: the one opening the connection (TLS client)

File: src/docs/asciidoc/security.adoc [hunk 4]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+and other one which accepts the connection (TLS server).
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: and the one accepting the connection (TLS server).

File: src/docs/asciidoc/security.adoc [hunk 5]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+By default only the TLS server proves its identity by presenting a certificate to the TLS client.
+The Mutual authentication means that also TLS clients prove their identity to TLS servers.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: .....
The mutual authentication means that also the TLS clients prove...

File: src/docs/asciidoc/security.adoc [hunk 6]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
 
-Starting with Hazelcast 3.8.1, mutual authentication is introduced. This allows the clients also to have their keyStores and members to have their trustStores so that the members can know which clients they can trust.
+Hazelcast members can be on both sides of TLS connection - TLS servers and TLS clients.
+Hazelcast clients are always on client side of a TLS connection.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: ...on the client side...

File: src/docs/asciidoc/security.adoc [hunk 7]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
 
-To enable mutual authentication, you need to set the following property at the member side:
+By default Hazelcast members have keyStore used to identify themselves to clients and other members.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: ...themselves to the clients and...

File: src/docs/asciidoc/security.adoc [hunk 8]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+Both Hazelcast members and Hazelcast clients have trustStore used to define which members they can trust.
 
-```
-props.setProperty("javax.net.ssl.mutualAuthentication", "REQUIRED");
-```
+When the Mutual authentication feature is enabled, Hazelcast clients need to have keyStore provided.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: When the mutual authentication feature is enabled, Hazelcast clients need to provide keyStore.

File: src/docs/asciidoc/security.adoc [hunk 9]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
+A client proves its idenity by providing its certificate to Hazelcast member it's connecting to.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: ...identity...

File: src/docs/asciidoc/security.adoc [hunk 10]
```diff
@@ -314,38 +345,40 @@ NOTE: When a new client is introduced with a new keystore, the truststore at the
...
+clientSslProps.setProperty("trustStore", "/opt/client.truststore");
+clientSslProps.setProperty("trustStorePassword", "123456");
+
 config.getNetworkConfig().setSSLConfig(new SSLConfig().setEnabled(true).setProperties(clientSslProps));
- 
 HazelcastClient.newHazelcastClient(config);
 ```
 
+If the mutual authentication is not required, the Hazelcast members will accept all incoming TLS connections without verifying if the connecting side is trusted.
+Therefor it's recommended to require the mutual authentication in Hazelcast members configuration.
```
Comment (Serdaro) [inline src/docs/asciidoc/security.adoc]: Therefore it's...

File: src/docs/asciidoc/security.adoc [hunk 11]
```diff
@@ -218,94 +219,124 @@ Hazelcast provides a default SSLContextFactory, `com.hazelcast.nio.ssl.BasicSSLC
...
 Here are the descriptions for the properties:
- 
-* `keystore`: Path of your keystore file. Note that your keystore's type must be `JKS`.
+
+* `keystore`: Path of your keystore file.
 * `keyStorePassword`: Password to access the key from your keystore file.
 * `keyManagerAlgorithm`: Name of the algorithm based on which the authentication keys are provided.
-* `keyStoreType`: The type of the keystore. Its default value is `JKS`. 
-* `truststore`: Path of your truststore file. The file truststore is a keystore file that contains a collection of certificates trusted by your application. Its type should be `JKS`.
+* `keyStoreType`: The type of the keystore. Its default value is `JKS`.
```
[the remark this answers is not in the document]
Comment (kwart) [reply src/docs/asciidoc/security.adoc]: Supported types depend on underlying Java.
We can mention the `PKCS12` which is a standard format and should be always available.

**Comment by kwart** [comment]
Thanks for the great review, @Serdaro. Could you recheck the PR? I've covered the comments.

File: src/docs/asciidoc/security.adoc (hunk 1 above)
Comment (Serdaro) [reply src/docs/asciidoc/security.adoc]: Actually there is kind of one: http://docs.hazelcast.org/docs/latest-dev/manual/html-single/index.html#configuring-cipher-suites

But you are right, out of scope here.

**Pull request merged**